### PR TITLE
iTrust Finance && Sashimiswap && KSwap finance && Updating masterchef.js

### PR DESCRIPTION
**Pull request by Mohammed18062020**
##### Twitter Link:
iTrust Finance -> https://twitter.com/itrustfinance?lang=en
Sashimiswap -> https://twitter.com/SASHIMISASHIMI5

##### Website Link:
iTrust Finance -> https://itrust.finance/
Sashimiswap -> https://sashimi.cool/home/swap

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
iTrust Finance -> 
image

Sashimiswap -> 
image

##### Current TVL:
iTrust Finance -> $11.52 M
Sashimiswap -> $14,735,158

##### Chain:
iTrust Finance -> ETHEREUM
Sashimiswap -> ETHEREUM, BSC and HECO

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)
iTrust Finance -> "id": "itrust-governance-token"
Sashimiswap -> "id": "sashimi"

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
iTrust Finance -> "id":11652
Sashimiswap -> "id":1175

##### Token address and ticker if any:
iTrust Token (ITG)-> 0xF88b137cfa667065955ABD17525e89EDCF4D6426
Sashimiswap (SASHIMI)  ->
     ETH: 0xC28E27870558cF22ADD83540d2126da2e4b464c2
     BSC: 0x97C5079e358FaBaf92a8B5D56473450142230194

##### Category (Yield/Dexes/Lending/Minting/Assets/Insurance/Options/Indexes/Staking) *Please choose only one:
iTrust Finance -> Insurance
Sashimiswap -> Dexes

### CODE REVIEW

**Comment by Mohammed18062020** [comment]
The masterchef.js file updating is for Stormswap protocol, because it's not included the "PGL" on the filter ```if(symbol.output.includes('LP') || symbol.output.includes('PGL'))```. The KSwap Protocol I updated it.

File: projects/iTrustfinance/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        target: vault,
+        ethBlock,
+      })
+    ).output;
+
+    sdk.util.sumSingleBalance(balances, NXM, supply);
+  }
+
+  for (const token of erc20Tokens) {
+    await sumTokensAndLPsSharedOwners(
```
Comment (0xngmi) [inline projects/iTrustfinance/index.js:49]: it's better to aggregate into a single call

File: projects/iTrustfinance/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+const vaults = [
+  // sNXM (Nexus Mutual Strategy Vault)
+  "0xD82c32Dd3585B5bc528688Efa408adA09963C238",
+  // iNXM (Nexus Mutual Index Vault)
+  "0xf4104CadE73d699922CC657A97CB1132bC620978",
+];
+
+/*
+ * WNXM and NXM have different price at coingecko! what do I should use?
```
Comment (0xngmi) [inline projects/iTrustfinance/index.js:13]: i'd just count them separately

File: projects/iTrustfinance/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  for (const vault of vaults) {
+    const supply = (
+      await sdk.api.abi.call({
+        abi: abi.totalSupply,
+        target: vault,
+        ethBlock,
+      })
+    ).output;
+
+    sdk.util.sumSingleBalance(balances, NXM, supply);
```
Comment (0xngmi) [inline projects/iTrustfinance/index.js:45]: supply = nxm balance?

File: projects/sashimiswap/index.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+const factoryHECO = "0xC28E27870558cF22ADD83540d2126da2e4b464c2";
+const comprollerHECO = "0x6Cb9d7ecf84b0d3E7704ed91046e16f9D45C00FA";
+const sIHT = "0xf13d3E10DEE31b80887422c89285112Dd00ce0B5";
+const WHTEquivalent = "0x5545153ccfca01fbd7dd11c0b23ba694d9509a6f";
+
+const staking = async (timestamp, ethBlock) => {
+  const balances = {};
+
+  const bal = (
+    await sdk.api.abi.call({
```
Comment (0xngmi) [inline projects/sashimiswap/index.js:29]: you can use ../helper/staking to simplify the code

File: projects/sashimiswap/index.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  return balances;
+};
+
+module.exports = {
+  staking: {
+    tvl: staking,
+  },
+  ethereum: {
+    tvl: ethTvl,
+    ethMarketsTvl,
```
Comment (0xngmi) [inline projects/sashimiswap/index.js:108]: this won't be used by the backend, will just be ignored

**Review by 0xngmi — CHANGES_REQUESTED** [review]

File: projects/iTrustfinance/index.js (hunk 3 above)
Comment (0xngmi) [reply projects/iTrustfinance/index.js:45]: this reply seems repeated

File: projects/kswapfinance/index.js [hunk 6]
```diff
@@ -1,75 +1,17 @@
...
+  "0xab0d1578216A545532882e420A8C61Ea07B00B12", // KST
+];
 
 module.exports = {
-  misrepresentedTokens: true,
-  tvl: sdk.util.sumChainTvls([okexTvl]),
+  tvl: calculateUsdUniTvl(factory, "okexchain", wokt, whitelist, "okexchain"),
   methodology:
-    "We count tvl on the mining pool threw LiquidityPool(pairs) contract and farming pool threw DepsitPool(single tokens) contract",
+    "We count tvl on LiquidityPool(pairs) and DepsitPool(single tokens) threw factory contract",
```
Comment (0xngmi) [inline projects/kswapfinance/index.js:16]: afaik this is not actually including single sided staking

Comment (Mohammed18062020) [reply projects/kswapfinance/index.js:16]: Yes, you are rigth, I forgot to update it!

File: projects/iTrustfinance/index.js (hunk 3 above)
Comment (Mohammed18062020) [reply projects/iTrustfinance/index.js:45]: Yes, because it is the respoune of both of doubts

**Pull request merged**